lib/spack/spack/s3_handler.py

```diff
@@ -23,11 +23,12 @@
 # https://github.com/python/cpython/pull/3249
 class WrapStream(BufferedReader):
     def __init__(self, raw):
-        raw.readable = lambda: True
-        raw.writable = lambda: False
-        raw.seekable = lambda: False
-        raw.closed = False
-        raw.flush = lambda: None
+        if not isinstance(raw, IOBase):
```

Comment:
Can you leave a comment about what version of boto lacks the interface?

Reply:
Done. I also added a link to the relevant commit in `botocore`.